src/coreclr/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyVersion.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.Loader
+{
+    internal struct AssemblyVersion : IEquatable<AssemblyVersion>
```

Comment:
Behavioral difference with `System.Version`: allows Unspecified value on each element.
Can be replaced together with AssemblyName replacement.